src/Microsoft.VisualStudio.Services.Agent/JobServer.cs

```diff
@@ -66,7 +67,19 @@ public async Task ConnectAsync(VssConnection jobConnection)
                 await Task.Delay(100);
             }
 
-            _taskClient = _connection.GetClient<TaskHttpClient>();
+            try
+            {
+                _taskClient = _connection.GetClient<TaskHttpClient>();
+            }
+            catch (SocketException e)
```

Comment:
Can we handle this more generic way - instead of creating try.. catch for all of these classes? Should we cover connection classes only?

Reply:
I think, no. I checked all possible call stacks for existed calling of HttpClient in agent code, and some cases require adding of new try-catch statements. Reason of this - some data, like server uri is unavailable in point with existed try-catch statements.
I think it's enough to cover HttpClient calling because SocketException can be thrown only during connection.

Reply:
Yes, per offline discussion also - moving this to more generic classes like VssHttpClientBase or creating a similar generic class, which allows traces output, would be some generic refactoring - probably it would be better to move it to the separate ticket and handle it currently this way.